pennylane/spin/lattice.py

```diff
@@ -358,21 +357,30 @@ def add_edge(self, edge_indices):
             self.edges.append(new_edge)
 
 
-def _generate_lattice(lattice, n_cells, boundary_condition=False, neighbour_order=1):
-    r"""Generates the lattice object for a given shape and n_cells.
+def generate_lattice(lattice, n_cells, boundary_condition=False, neighbour_order=1):
+    r"""Generates a Lattice object for a given lattice shape and number of cells.
 
     Args:
         lattice (str): Shape of the lattice. Input values can be ``'chain'``, ``'square'``, ``'rectangle'``,
-              ``'honeycomb'``, ``'triangle'``, ``'kagome'``, ``'lieb'``, ``'cubic'``, ``'bcc'``, ``'fcc'``,
-               or ``'diamond'``.
+            ``'honeycomb'``, ``'triangle'``, ``'kagome'``, ``'lieb'``, ``'cubic'``, ``'bcc'``, ``'fcc'``,
+            or ``'diamond'``.
         n_cells (list[int]): Number of cells in each direction of the grid.
         boundary_condition (bool or list[bool]): Defines boundary conditions in different lattice axes.
-                               Default is ``False`` indicating open boundary condition.
+            Default is ``False`` indicating open boundary condition.
         neighbour_order (int): Specifies the interaction level for neighbors within the lattice.
-                               Default is 1, indicating nearest neighbour.
+            Default is 1, indicating nearest neighbour.
 
     Returns:
         lattice object.
```

Comment:
Might be nice to add a return type here.

Reply:
Added.